src/export/Package.ts

```diff
@@ -30,85 +30,123 @@ export class Package implements Fishable {
       ];
     }
 
+    // version needs to be checked separately from the base
+    const [base, ...versionParts] = item?.split('|') ?? ['', ''];
+    const version = versionParts.join('|') || null;
+
     for (const type of types) {
       let def;
       switch (type) {
         case Type.Profile:
-          def = this.profiles.find(p => p.id === item || p.name === item || p.url === item);
+          def = this.profiles.find(
+            p =>
+              (p.id === base || p.name === base || p.url === base) &&
+              (version == null || p.version === version)
+          );
           if (!def) {
             def = this.instances.find(
               i =>
                 i._instanceMeta.usage === 'Definition' &&
                 i.resourceType === 'StructureDefinition' &&
                 i.derivation === 'constraint' &&
                 i.type !== 'Extension' &&
-                (i.id === item || i._instanceMeta.name === item || i.url === item)
+                (i.id === base || i._instanceMeta.name === base || i.url === base) &&
+                (version == null || (i.version && i.version === version))
```

Comment:
Well... I thought there was a reason, but now I can't think of what that reason was.

Reply:
I think `(version == null || (i.version && i.version === version))` can be shortened to `(version == null || i.version === version)` because:
* If `version` is null or undefined, it is true without need to evaluate the 2nd part;
* If `version` is not null/undefined, then it won't be equal to `i.version` if `i.version` is null anyway.

Reply:
Yeah agreed. Maybe I had something different at some point and changed things around and didn't realize the redundancy in the end.

Reply:
Addressed in [9668cdc](https://github.com/FHIR/sushi/pull/1275/commits/9668cdc2e8174378a573a33170c4740424a1b2cd)